### PR TITLE
coprocessor, util: reorganize xeval, codec dirs

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR re-organizes `xeval`, `codec` directories. It moves these dirs from  dir `util` into dir `coprocessor`. And it contains only modification to the import paths.

PTAL @AndreMouche @lishihai9017 @andelf

### CODE REVIEW

**Comment by hhkbp2** [comment]
PTAL @siddontang

File: src/coprocessor/executor/aggregation.rs [hunk 1]
```diff
@@ -21,12 +21,12 @@ use tipb::executor::Aggregation;
 use tipb::expression::Expr;
 
 use util::collections::{HashMap, HashMapEntry as Entry};
-use util::codec::datum::{self, DatumEncoder, approximate_size};
-use util::codec::table::RowColsDict;
-use util::xeval::{Evaluator, EvalContext};
 
 use super::super::Result;
 use super::{Executor, Row, ExprColumnRefVisitor};
```
Comment (AndreMouche) [inline src/coprocessor/executor/aggregation.rs]: Move this line before L25?

File: src/coprocessor/aggregate.rs [hunk 2]
```diff
@@ -14,9 +14,8 @@
 use std::cmp::Ordering;
```
Comment (AndreMouche) [inline src/coprocessor/aggregate.rs]: Could we move  this file in directory executor?

File: src/coprocessor/executor/aggregation.rs (hunk 1 above)
Comment (hhkbp2) [inline src/coprocessor/executor/aggregation.rs]: I'm not sure whether there is rule defined in rust for organizing these import paths.

This order of putting further paths ahead like fine to me, as long as it's used consistently:
```
use super::super::XXX;
use super::XXX::xxx;
use super::xxx;
```

File: src/coprocessor/aggregate.rs (hunk 2 above)
Comment (hhkbp2) [inline src/coprocessor/aggregate.rs]: I would like to keep all the code in `coprocessor` unchanged in this PR.

The old aggregate code should not enter executor directory IMO. It's not one of the new DAG executor and should be removed some time later.

File: src/coprocessor/executor/aggregation.rs (hunk 1 above)
Comment (siddontang) [inline src/coprocessor/executor/aggregation.rs]: /cc @andelf

**Comment by siddontang** [comment]
It is fine to me. 
PTAL @andelf

File: src/coprocessor/codec/mysql/json/path_expr.rs [hunk 3]
```diff
@@ -40,7 +40,7 @@
 use std::ops::Index;
 use std::ascii::AsciiExt;
 use regex::Regex;
-use super::super::Result;
+use super::super::super::Result;
```
Comment (andelf) [inline src/coprocessor/codec/mysql/json/path_expr.rs]: 😞 refer this from top level?

Comment (hhkbp2) [inline src/coprocessor/codec/mysql/json/path_expr.rs]: ok. then.

Comment (hhkbp2) [inline src/coprocessor/codec/mysql/json/path_expr.rs]: address comment.

**Comment by hhkbp2** [comment]
PTAL @andelf

**Comment by hhkbp2** [comment]
PTAL @lishihai9017

**Comment by lishihai9017** [comment]
LGTM

**Comment by hhkbp2** [comment]
@andelf There are a few PRs which rely on this PR. I need to get this PR merged so that these PRs could get rolling ASAP. Comment this PR if you see any issue later when you have time, I will open a new PR to fix them if necessary.

**Pull request merged**